src/webui/www/private/views/search.html

```diff
@@ -462,36 +460,43 @@
             const populateCategorySelect = function(categories) {
                 const categoryHtml = [];
                 categories.each(function(category) {
-                    categoryHtml.push("<option>" + category + "</option>");
+                    const option = new Element("option");
+                    option.set("value", category.id);
+                    option.set("html", category.name);
+                    categoryHtml.push(option.outerHTML);
                 });
 
                 // first category is "All Categories"
-                if (categoryHtml.length > 1)
+                if (categoryHtml.length > 1) {
                     // add separator
-                    categoryHtml.splice(1, 0, "<option disabled>──────────</option>");
+                    const option = new Element("option");
+                    option.set("disabled", true);
+                    option.set("html", "──────────");
+                    categoryHtml.splice(1, 0, option.outerHTML);
+                }
 
                 $('categorySelect').set('html', categoryHtml.join(""));
             };
 
             const selectedPlugin = $('pluginsSelect').get("value");
-            if ((selectedPlugin === "all") || (selectedPlugin === "enabled")) {
-                const url = new URI('api/v2/search/categories');
-                url.setData('name', selectedPlugin);
-                new Request.JSON({
-                    url: url,
-                    noCache: true,
-                    method: 'get',
-                    onSuccess: function(response) {
-                        populateCategorySelect(response);
-                    }
-                }).send();
+
+            if (selectedPlugin === "all" || selectedPlugin === "enabled") {
+                const categoriesSet = new Set();
+                const categories = []
+                for (const plugin of searchPlugins) {
+                    if (selectedPlugin === "enabled" && !plugin.enabled) continue
+                    for (const category of plugin.supportedCategories) {
+                        if (!categoriesSet.has(category.id)) {
+                            categoriesSet.add(category.id)
+                            categories.push(category)
+                        }
+                     }
+                }
+                populateCategorySelect(categories)
```

Comment:
Let's extract `categories` from `categoriesSet` instead:
```js
const uniqueCategories = new Set();  // maybe rename the variable?
for (const plugin of searchPlugins) {
    // ...
}
const categories = []
uniqueCategories.forEach(category => categories.push(category));
```

Reply:
This doesn't directly work because the set has to hold category ids (due to object equality), but I was able to rework the logic to match your pattern. Good catch.

```js
const uniqueCategories = {};
for (const plugin of searchPlugins) {
    // ...
}
const categories = Object.keys(uniqueCategories).sort().map(id => uniqueCategories[id]);
populateCategorySelect(categories);
```